core/trino-main/src/test/java/io/trino/util/TestFastutilSetHelper.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.util;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeOperators;
+import it.unimi.dsi.fastutil.longs.AbstractLongSet;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Set;
+
+import static io.trino.spi.function.InvocationConvention.InvocationArgumentConvention.NEVER_NULL;
+import static io.trino.spi.function.InvocationConvention.InvocationReturnConvention.FAIL_ON_NULL;
+import static io.trino.spi.function.InvocationConvention.InvocationReturnConvention.NULLABLE_RETURN;
+import static io.trino.spi.function.InvocationConvention.simpleConvention;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.DateType.DATE;
+import static io.trino.spi.type.DecimalType.createDecimalType;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.RealType.REAL;
+import static io.trino.spi.type.SmallintType.SMALLINT;
+import static io.trino.spi.type.TimeType.TIME_MICROS;
+import static io.trino.spi.type.TimeType.TIME_MILLIS;
+import static io.trino.spi.type.TimeType.TIME_NANOS;
+import static io.trino.spi.type.TimeType.TIME_PICOS;
+import static io.trino.spi.type.TimeType.TIME_SECONDS;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_MICROS;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_MILLIS;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_SECONDS;
+import static io.trino.util.FastutilSetHelper.toFastutilHashSet;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestFastutilSetHelper
+{
+    private static final TypeOperators TYPE_OPERATORS = new TypeOperators();
+
+    @Test
+    public void testCreateTupleDomainFilters()
```

Comment:
Remove "public" -- it's the convention for JUnit tests.

Reply:
Almost all of our existing code base is still using the old convention of "public" for tests.
Do we want to create inconsistency in the code by having some not "public" among mostly "public" tests ?
IMO it would be better to do a one time re-write of all the tests and use code style check to enforce this rather than manually enforcing this new convention on every incoming PR.

Reply:
Changing them all in one shot is going to be impossible. This is also not something that can be enforced via code style. Sometimes, they need to be public due to using inheritance to override tests (although that's a different story to be addressed at some point)